libethcore/Block.cpp

```diff
@@ -25,30 +25,42 @@
 #include "TxsParallelParser.h"
 #include <libdevcore/Guards.h>
 #include <libdevcore/RLP.h>
-#include <libdevcore/easylog.h>
 #include <tbb/parallel_for.h>
 
+
+#define BLOCK_LOG(LEVEL)    \
+    LOG(LEVEL) << "[Block]" \
+               << "[line:" << __LINE__ << "]"
+
 namespace dev
 {
 namespace eth
 {
 Block::Block(
     bytesConstRef _data, CheckTransaction const _option, bool _withReceipt, bool _withTxHash)
 {
+    m_transactions = std::make_shared<Transactions>();
+    m_transactionReceipts = std::make_shared<TransactionReceipts>();
+    m_sigList = std::make_shared<std::vector<std::pair<u256, Signature>>>();
+
     decode(_data, _option, _withReceipt, _withTxHash);
 }
 
 Block::Block(
     bytes const& _data, CheckTransaction const _option, bool _withReceipt, bool _withTxHash)
 {
+    m_transactions = std::make_shared<Transactions>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'm_transactions' is assigned in constructor body. Consider performing initialization in initialization list.](https://app.codacy.com/app/fisco/FISCO-BCOS/pullRequest?prid=4732332)